src/main/java/ipman/ui/CliUi.java

```diff
@@ -0,0 +1,53 @@
+package ipman.ui;
+
+import java.util.Scanner;
+
+import ipman.commands.Command;
+import ipman.parser.Parser;
+
+/**
+ * Represents the command line user interface.
+ * Messages will be received from standard input and messages will be displayed
+ * to standard output.
+ */
+public class CliUi implements Ui {
+    private static final String HORIZONTAL_LINE = "____________________________________________________________";
+    private final Scanner scanner = new Scanner(System.in);
+
+    @Override
+    public void showWelcome() {
+        System.out.println(HORIZONTAL_LINE);
+        System.out.println("Hello! I'm Ip Man.");
+        System.out.println("What can I do for you?");
+    }
+
+    @Override
+    public Command readMessage() {
+        System.out.print("> ");
+        String message = scanner.nextLine();
+        return Parser.parseCommand(message);
+    }
+
+    @Override
+    public void showMessage(String message) {
+        System.out.println(message);
+    }
+
+    @Override
+    public void showMessageF(String message, Object... args) {
```

Comment:
Agreed that a better method name could be used in this case!

Reply:
Resolved by https://github.com/nus-cs2103-AY2425S1/ip/pull/538/commits/338c4f7e995a96a70ec07341c163301fa9ab3e75